lib/input_data_page.dart

```diff
@@ -0,0 +1,118 @@
+                if (state is InputDataInitialState) {
+                  return Column(children: [
+                    datePick(Constants.fromDateNameString,
+                        DateTime(Constants.defaultDate)),
+                    const SizedBox(height: 15),
+                    datePick(Constants.toDateNameString, state.lastDate),
+                  ]);
+                } else if (state is PickedDateState) {
+                  return Column(children: [
+                    datePick(
+                        Constants.fromDateNameString, state.dateOfBirth),
+                    const SizedBox(height: 15),
+                    datePick(Constants.toDateNameString, state.lastDate),
+                  ]);
+                } else {
+                  return const Text('Unknown error');
+                }
+              }),
+          const SizedBox(height: 15),
+          ElevatedButton(
+            onPressed: () {
+              if (_bloc.dateOfBirth != DateTime(0000)) {
+                _bloc.calculateAge();
+                Navigator.push(
+                    context,
+                    MaterialPageRoute(
+                        builder: (_) => BlocProvider.value(
+                              value: _bloc,
+                              child: CalculatedDataPage(),
+                            )));
+              } else {
+                showPicker(Constants.fromDateNameString);
+              }
+            },
+            child: const Text('Calculate Age'),
+          )
+        ],
+      ),
+    );
+  }
+
+  Widget datePick(String widgetName, DateTime pick) {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(widgetName, style: const TextStyle(fontSize: 17)),
+        Container(
+          decoration: BoxDecoration(
+            border: Border.all(
+              color: Colors.black,
+              width: 3,
+            ),
+            borderRadius: BorderRadius.circular(12),
+          ),
+          child: Padding(
+            padding: const EdgeInsets.all(8.0),
+            child: Row(
+              children: [
+                const SizedBox(
+                  width: 25,
+                ),
+                InkWell(
+                    onTap: () {
+                      showPicker(widgetName).then((date) {
+                        if (widgetName == Constants.fromDateNameString) {
+                          _bloc.addFirstDateOfBirthEvent(date!);
+                        } else if (widgetName == Constants.toDateNameString) {
+                          _bloc.addSecondDateOfBirthEvent(date!);
+                        }
+                      });
+                    },
+                    child: Text(pick == DateTime(Constants.defaultDate)
+                        ? 'Pick up a date'
+                        : pick.toString())),
+              ],
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+
+  Future<DateTime?> showPicker(String widgetName) {
```

Comment:
Move to above-mentioned new widget.
Let this function be private

Reply:
showPicker moved to DatePick class and marked private
on raw #56 changed to show just SnackBar